### PR TITLE
Added IDP initiated flow flag in ParseHashOptions  (__enableIdPInitiatedLogin)

### PR DESCRIPTION
**Pull request by Tanver-Hasan**
Please fill in this template.

- [ ] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [ ] Add or edit tests to reflect the change. (Run with `npm test`.)
- [ ] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [ ] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

If removing a declaration:
- [ ] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Comment by marknelissen** [comment]
Can you please provide a link to the documentation mentioning the additional option?

**Comment by peterblazejewicz** [comment]
> Can you please provide a link to the documentation mentioning the additional option?

> __enableIdPInitiatedLogin

https://github.com/auth0/docs/pull/5857/files
https://auth0.com/docs/protocols/saml/idp-initiated-sso
https://github.com/auth0/auth0.js/commit/3c60563717fd2de526ea162c0a40ab39aba172ea

File: types/auth0-js/index.d.ts [hunk 1]
```diff
@@ -747,6 +747,8 @@ export interface ParseHashOptions {
     state?: string;
     nonce?: string;
     _idTokenVerification?: boolean;
+    __enableIdPInitiatedLogin?: boolean;
+
```
Comment (peterblazejewicz) [inline types/auth0-js/index.d.ts]: You'd have to fix this to pass the CI build:

> Error: /home/travis/build/DefinitelyTyped/DefinitelyTyped/types/auth0-js/index.d.ts:752:1
> ERROR: 752:1  no-padding  Don't leave a blank line before '}'. See: https://github.com/Microsoft/dtslint/blob/master/docs/no-padding.md

**Review by peterblazejewicz — APPROVED** [review]
:shipit:  after fixing CI

**Pull request closed without merging**